base/parse.jl

```diff
@@ -194,3 +194,6 @@ function parse(str::AbstractString; raise::Bool=true)
     end
     return ex
 end
+
+promote_op{R<:AbstractFloat,S<:AbstractString}(::typeof(parse), ::Type{R}, ::Type{S}) = R
```

Comment:
I though it was practical to use this when the return type is part of the arguments so you know you are already expecting a specific type. I wasn't thinking in doing anything like this in any other case.